addons/mail/static/src/chatter/web/chatter_patch.js

```diff
@@ -188,24 +170,26 @@ patch(Chatter.prototype, {
     },
 
     async updateRecipients(record, mode = this.state.composerType) {
-        if (!record) {
+        if (!record || !this.state.thread) {
             return;
         }
         const partnerIds = []; // Ensure that we don't have duplicates
         let email;
-        this.mailImpactingFields.recordFields.forEach((field) => {
-            const value = record._changes[field];
-            if (record.data[field] !== undefined && value) {
-                partnerIds.push(value[0]);
-            }
-        });
-        this.mailImpactingFields.emailFields.forEach((field) => {
-            const value = record._changes[field];
-            if (record.data[field] !== undefined && value) {
+        if (this.state.thread.partner_fields) {
```

Comment:
Don't forget to define the new fields in thread model + d.ts for proper doc + auto complete.